diff --git a/dist/fromFile.cjs b/dist/fromFile.cjs
index c226d3dac07f235db456c774b26cca88f655e1af..4d50371fb963d0374060b7156cd0cdb2bd845585 100644
--- a/dist/fromFile.cjs
+++ b/dist/fromFile.cjs
@@ -1036,7 +1036,7 @@ var processQueue = async () => {
       }
       const inputSize = Math.min(size, MaxInputSize);
       const input = new Uint8Array(inputSize);
-      await handle.read(input, 0, inputSize, 0);
+      await handle.read(Buffer.from(input), 0, inputSize, 0);
       resolve2(imageSize(input));
     } catch (err) {
       reject(err);
